text/0000-rtt-computation.md

```diff
@@ -0,0 +1,119 @@
+# RTT computation. A mechanism to compute and expose round-trip times
+[feature]: #feature
+
+- Start Date: `2021-07-08`
+- RFC PR: [#0000](https://github.com/Unity-Technologies/com.unity.multiplayer.rfcs/pull/0000)
+- SDK PR: [#0000](https://github.com/Unity-Technologies/com.unity.multiplayer.mlapi/pull/0000)
+
+# Summary
+[summary]: #summary
+
+This would add an API for game code to read measured round-trip times, both on the server and client. The round-trip time exposed would be the high-level RTT: How much time goes by between the moment a high-level system writes something to the network and the moment it gets a response from the other machine. 
+
+This is distinct from transport-level RTT which captures only the latency at the network level.
+
+This RFC provides an API to read the RTT, and a mechanism for high-level systems to drive it. It doesn't describe the first usage of this system.
+
+# Motivation
+[motivation]: #motivation
+
+High-level systems, like snapshotting, interpolation, and client-side prediction often need to know when an action performed locally could/will be surfaced on the other machines. Game code might also be interested in the overall latency a game sees.   
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+When game code needs to read latency it checks in `public NetworkManager.ConnectedClientsList[index].GetRtt();`. This new function returns
+
+```
+public struct Rtt
+{
+    public double best;
+    public double average;
+    public double worst;
+    public int sampleCount;
+}
+```
+
+The members of this struct describe, for a given client, the RTT. `best` shows the best RTT measured, `average` shows the average RTT measured, and `worst` shows the worst RTT measured. `sampleCount` indicates how many samples were used to get those values. A count of 0 indicates no data is available. In this case, the other fields would be 0.
```

Comment:
For average we could consider using an exponential moving average which usually gives a better result.

Reply:
Yes. Good idea. I would just argue, then, to add a `internal void Reset()` to `NetworkConnection`, because exponential moving average keeps contributions forever (with decreasing weight, but still).